sdk/signature-mpc-wasm/src/lib.rs

```diff
@@ -213,10 +228,50 @@ pub fn recovery_id_sha256(
     })?;
     let signature: SignatureK256Secp256k1 = bcs::from_bytes(&signature)?;
 
-    Ok(recovery_id(message, public_key, signature, &Hash::SHA256).map_err(|_| JsErr {
-        message: "Can't generate RecoveryId".to_string(),
-        display: "Can't generate RecoveryId".to_string(),
-    })?.into())
+    Ok(recovery_id(message, public_key, signature, &Hash::SHA256)
+        .map_err(|_| JsErr {
+            message: "Can't generate RecoveryId".to_string(),
+            display: "Can't generate RecoveryId".to_string(),
+        })?
+        .into())
+}
+
+#[wasm_bindgen]
+pub fn generate_keypair() -> Result<JsValue, JsErr> {
+    let (public_key, private_key) =
+        signature_mpc::twopc_mpc_protocols::encrypt_user_share::generate_keypair()
+            .map_err(|e| to_js_err(e))?;
+    Ok(serde_wasm_bindgen::to_value(&(public_key, private_key))?)
+}
+
+#[wasm_bindgen]
+pub fn generate_proof(secret_share: Vec<u8>, public_key: Vec<u8>) -> Result<JsValue, JsErr> {
```

Comment:
Could you please clarify where this should be used? I'm referring to [this test](https://github.com/dwallet-labs/enhanced-maurer/blob/main/src/proof.rs#L450) as an example. Can you guide me on the appropriate usage or location for it? Additionally, we've not used this until now, and the verification has been passing successfully.

Reply:
@ycscaly why?

Reply:
Because otherwise how will users use it? They don't have access to the secret share, only the DKG ouptut

Reply:
We already talked about it.
Currently this code is used like this:

```typescript
		const dwallet = await createDWallet(toolbox.keypair, toolbox.client);
		const dwalletID = dwallet?.dwalletId!;
		// Extract the secret share from the DKGOutput.
		// It's the first 32 bytes because it's the first field in the Output Rust struct.
		let secretShareBytes = dwallet?.dkgOutput!.slice(0, 32);
		let secretShare = new Uint8Array(secretShareBytes);
		const encryptedUserShareAndProof = generate_proof(secretShare, recipientData?.encryptionKey!);
```

The `secretShare` is taken from the DKG output.
If we pass only the DKG output then we will need to do the same thing as above, but in Rust.

In order to avoid it we have to move the `signature_mpc::twopc_mpc_protocols::encrypt_user_share::generate_proof` code to the 2PCMPC lib.

I think for now let's keep it and if we decide to move the code to the 2PCMPC lib, then we can change this code appropriately.

Reply:
https://github.com/dwallet-labs/2pc-mpc/pull/26 approved, now you can fix it so that there are no weird bypasses. Make the rust code take `dkg_output`, or create a new wasm function that takes the `secret_key_share` out of it.